lib/rucio/core/transfer.py

```diff
@@ -664,15 +664,15 @@ def __rewrite_source_url(source_url, source_sign_url, dest_sign_url, source_sche
         if source_scheme in ['davs', 'https']:
             source_url += '?copy_mode=%s' % WEBDAV_TRANSFER_MODE
 
-    if source_sign_url == 'gcs':
-        source_url = re.sub('davs', 'gclouds', source_url)
-        source_url = re.sub('https', 'gclouds', source_url)
-    elif source_sign_url == 's3':
-        source_url = re.sub('davs', 's3s', source_url)
-        source_url = re.sub('https', 's3s', source_url)
-
-    if source_scheme == 'https+srm':
-        source_url = re.sub('srm', 'https+srm', source_url)
+    source_sign_url_map = {'gcs': 'gclouds', 's3': 's3s'}
+    if source_sign_url in source_sign_url_map:
+        if source_url[:7] == 'davs://':
+            source_url = source_sign_url_map[source_sign_url] + source_url[4:]
+        if source_url[:8] == 'https://':
+            source_url = source_sign_url_map[source_sign_url] + source_url[5:]
```

Comment:
I find the new organisation with a map; and iteration over it; much too difficult to follow for the little benefit it bring. If we don't intend to add many more protocols in the future to this map, it's probably better to keep it explicit and easy.

Reply:
That being said, I'm not against to merge it without changes.